xdsl/dialects/llvm.py

```diff
@@ -1170,6 +1171,14 @@ def print_parameter(self, printer: Printer):
                 )
 
 
+@irdl_attr_definition
+class FastMathAttr(FastMathAttrBase):
+    name = "llvm.fastmath"
+
+    def __init__(self, flags: None | Sequence[FastMathFlag] | Literal["none", "fast"]):
+        super().__init__(flags)
+
```

Comment:
I think you can just remove it then?

Reply:
Actually, no, but I don't know why ^^'
My guess is that I'm doing something wrong in the `irdl_op_definition`, but I'm not really sure why?
I'm adding an issue on this though, as this *should* work

Reply:
I left a comment on the issue, I think `Data.new` always calls init on `Data`, not on the class

Reply:
Feels like we might as well fix that before merging this

Reply:
I would suggest to not PR creep this here.
The solution is a bit more complex than expected.

The issue is that irdl_op_definition generates the `__init__`, as someone might inherit an `ABC` `Attribute` and then add more fields, so we cannot use the `super().__init__` by default. We would need to check that there is a previous `__init__`, and that no more fields are added in the child class.
So I would like to not do that here, and do this later when this become relevant (especially if we change the way we handle attributes)

Reply:
fair enough

Reply:
We should probably try and add some docs to explain the workaround